airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -846,15 +846,14 @@ def delete_objects(self, bucket: str, keys: Union[str, list]) -> None:
     @provide_bucket_name
     @unify_bucket_name_and_key
     def download_file(
-        self, key: str, bucket_name: Optional[str] = None, local_path: Optional[str] = None
+        self, key: str, bucket_name: Optional[str] = None, filename: Optional[str] = None
```

Comment:
Filename is made optional here with `None` as default value. It seems filename cannot be `None` as per Boto S3 docs. What happens when `filename` is not passed?

This is also a change in the documented interface and might need changelog update : https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/_api/airflow/providers/amazon/aws/hooks/s3/index.html?highlight=download_file#airflow.providers.amazon.aws.hooks.s3.S3Hook.download_file

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.download_file

Reply:
Thanks for the review. 

Good catch on the filename - I will make remove the option for that. 

Not sure where to edit the docs for the s3_hook, the link at the bottom of the page:  https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/_api/airflow/providers/amazon/aws/hooks/s3/index.html  that reads: "Suggest a change on this page" gives me a 404 error: https://github.com/apache/airflow/edit/main/docs/apache-airflow-providers-amazon/_api/airflow/providers/amazon/aws/hooks/s3/index.rst

Reply:
Good point. In that case we can keep the argument `local_path`.
If a local_path variable is a path to a directory, then we run `s3.download_fileobj`. If its a path to a file, then we run `s3.download_file`. I think this workflow would be non-breaking. 
We would still have to update the docs to reflect this change though.

Reply:
don't get me wrong.. if you think it's best to use the new name we can do the deprecation..
it's not that complex it just add 3-4 code lines :) 
See for example:

https://github.com/apache/airflow/blob/ccb5ce934cd521dc3af74b83623ca0843211be62/airflow/providers/trino/hooks/trino.py#L165-L173

Reply:
Yeah. Deprecations should be when we have some external circumstances - such as changing naming conventions from libraries etc. But if we are in full control of the API and can choose, then it's better to keep it backwards compatible.

Reply:
Hi! I have updated this pull request based on the feedback above. Thanks so much everyone. Hope this version is okay. 
I've added provision for using local_path as path to a directory or a file.